src/multi/bmp180.cpp

```diff
@@ -0,0 +1,305 @@
+{
+  return std::to_array<hal::byte>({ 0xAA });
+}
+
+auto control_measurement_register()
+{
+  return std::to_array<hal::byte>({ 0xF4 });
+}
+
+auto uncompensated_output_data_register()
+{
+  return std::to_array<hal::byte>({ 0xF6 });
+}
+
+auto soft_reset_register()
+{
+  return std::to_array<hal::byte>({ 0xE0 });
+}
+
+// struct to hold the computation variables
+struct computation_variables
+{
+  std::int32_t x1, x2, x3;
+  std::int32_t b3, b5, b6;
+  std::uint32_t b4, b7;
+  std::int32_t temperature, pressure;
+};
+
+// struct to hold temperature helper function results
+struct temperature_results
+{
+  hal::celsius temperature;
+  std::int32_t b5;
+};
+
+// helper function to check if sensor has finished its data conversion
+int conversion_status(hal::i2c* p_i2c, hal::byte p_address)
+{
+  std::array<hal::byte, 1> buffer;
+  hal::write_then_read(*p_i2c,
+                       p_address,
+                       control_measurement_register(),
+                       buffer,
+                       hal::never_timeout());
+  hal::byte check_bit = ((buffer[0]) >> 5) & 1;
+  if (check_bit == 0)
+    return 1;
+  else
+    return 0;
+}
+
+// helper function that computes and returns both the temperature result, as
+// well as the b5 variable for pressure() to use
+temperature_results compute_temperature_and_b5(hal::i2c* p_i2c,
+                                               hal::byte p_address,
+                                               std::uint16_t p_ac5,
+                                               std::uint16_t p_ac6,
+                                               std::int16_t p_mc,
+                                               std::int16_t p_md)
+{
+  computation_variables variables;
+
+  // tell sensor to start conversion for temperature
+  hal::byte control_setting = 0x2E;
+  std::array<hal::byte, 2> sensor_config_buffer = {
+    control_measurement_register()[0], control_setting
+  };
+  hal::write(*p_i2c, p_address, sensor_config_buffer, hal::never_timeout());
+
+  // loop to wait for conversion to complete
+  int conversion_finished = 0;
+  while (conversion_finished == 0) {
+    conversion_finished = conversion_status(p_i2c, p_address);
+  }
+
+  // grab the raw data from the conversion and store it
+  auto uncompensated_data_buffer =
+    hal::write_then_read<2>(*p_i2c,
+                            p_address,
+                            uncompensated_output_data_register(),
+                            hal::never_timeout());
+  std::int32_t uncompensated_temperature =
+    (uncompensated_data_buffer[0] << 8) | uncompensated_data_buffer[1];
+
+  // compute the temperature in steps of 0.1C
+  variables.x1 = ((uncompensated_temperature - p_ac6) * p_ac5) >> 15;
+  variables.x2 = (p_mc << 11) / (variables.x1 + p_md);
+  variables.b5 = variables.x1 + variables.x2;
+  variables.temperature = (variables.b5 + 8) >> 4;
+
+  hal::celsius temperature_in_celsius =
+    static_cast<float>(variables.temperature) / 10.0;
```

Comment:
```suggestion
  hal::celsius temperature_in_celsius =
    static_cast<float>(variables.temperature) / 10.0f;
```

Use suffix f to make sure a number is float not double.